wb2k/__main__.py

```diff
@@ -109,13 +111,35 @@ def cli(channel, verbose):
         logger.debug("Found channel ID {} for #{}".format(channel_id, channel))
 
         logger.info("Listening for joins in #{}".format(channel))
-        while True:
 
-            # Handle dem messages!
-            for message in sc.rtm_read():
-                handle_message(message, channel, channel_id, sc, logger)
+        retry_count = 0
+        backoff = 0.5
 
-            time.sleep(0.5)
+        while True:
+            try:
+                # Handle dem messages!
+                for message in sc.rtm_read():
+                    handle_message(message, channel, channel_id, sc, logger)
+
+                # reset exponential backoff retry strategy every time we successfully loop
+                # failure would have happened at rtm_read call
+                retry_count = 0
+
+                time.sleep(0.5)
+            except WebSocketConnectionClosedException:
+                logger.error("Lost connection to Slack, reconnecting...")
+                if not sc.rtm_connect():
+                    logger.info("Failed to reconnect to Slack")
+                    if retry_count >= retries:
+                        sys.exit(bail('fatal', 'red', "Too many failed reconnect attempts, shutting down"))
+                    time.sleep((backoff ** 2) / 4)
+                else:
+                    logger.info("Reconnected to Slack")
+
+                retry_count += 1
 
     else:
         sys.exit(bail('fatal', 'red', "Couldn't connect to Slack"))
+
+if __name__ == '__main__':
```

Comment:
This allows you to clone the repo, then run `python wb2k` and have everything start.